lang/cs/Org.Apache.REEF.Client/Common/JavaClientLauncher.cs

```diff
@@ -0,0 +1,124 @@
+﻿/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using Org.Apache.REEF.Client.API.Exceptions;
+using Org.Apache.REEF.Tang.Annotations;
+
+namespace Org.Apache.REEF.Client.Common
+{
+    /// <summary>
+    ///     Helper class to launch the java side of the various clients.
+    /// </summary>
+    internal class JavaClientLauncher
+    {
+        /// <summary>
+        ///     The folder in which we search for the client jar.
+        /// </summary>
+        private const string JarFolder = "./";
+
+        [Inject]
+        private JavaClientLauncher()
+        {
+        }
+
+        internal void Launch(string javaClassName, params string[] parameters)
+        {
+            IList<string> arguments = new List<string>();
+            arguments.Add("-cp");
+            arguments.Add(GetClientClasspath());
+            arguments.Add(javaClassName);
+            foreach (var parameter in parameters)
+            {
+                arguments.Add(parameter);
+            }
+
+            var process = Process.Start(GetJavaCommand(), string.Join(" ", arguments));
+            if (process != null)
+            {
+                process.WaitForExit();
+            }
+            else
+            {
+                throw new Exception("Java client process didn't start.");
```

Comment:
We have utilities for exceptions. Org.Apache.REEF.Utilities.Diagnostics.Exceptions.Throw